src/app/profile/my-spaces/my-spaces.component.spec.ts

```diff
@@ -92,29 +92,103 @@ describe('MySpacesComponent', () => {
-        { provide: UserService, useValue: mockUserService },
-        { provide: AuthenticationService, useValue: mockAuthenticationService },
-        { provide: Broadcaster, useValue: mockBroadcaster }
+        {
+          provide: Contexts,
+          useFactory: (): jasmine.SpyObj<Contexts> => {
+            const mockContexts: jasmine.SpyObj<Contexts> = createMock(Contexts);
+            mockContexts.current = of(mockContext) as Observable<Context> & jasmine.Spy;
+            return mockContexts;
+          }
+        },
+        {
+          provide: GettingStartedService,
+          useFactory: (): jasmine.SpyObj<GettingStartedService> => {
+            const mockGettingStartedService: jasmine.SpyObj<GettingStartedService> = createMock(GettingStartedService);
+            mockGettingStartedService.createTransientProfile.and.returnValue(mockExtProfile);
+            mockGettingStartedService.update.and.returnValue(of({}));
+            return mockGettingStartedService;
+          }
+        },
+        {
+          provide: Logger,
+          useFactory: (): jasmine.SpyObj<Logger> => {
+            const mockLogger: jasmine.SpyObj<Logger> = createMock(Logger);
+            mockLogger.error.and.stub();
+            return mockLogger;
+          }
+        },
+        {
+          provide: Broadcaster,
+          useFactory: (): jasmine.SpyObj<Broadcaster> => {
+            const mockBroadcaster: jasmine.SpyObj<Broadcaster> = createMock(Broadcaster);
+            mockBroadcaster.on.and.callFake((key: string): Observable<any> => {
+              if (key === 'displayMySpaces') {
+                return of([spaceMock1, spaceMock2]);
+              }
+              if (key === 'displaySharedSpaces') {
+                return of([spaceMock1, spaceMock2]);
+              }
+            });
+            mockBroadcaster.broadcast.and.callThrough();
+            return mockBroadcaster;
+          }
+        },
+        {
+          provide: BsModalService,
+          useFactory: (): jasmine.SpyObj<BsModalService> => {
+            const mockBsModalService: jasmine.SpyObj<BsModalService> = createMock(BsModalService);
+            mockBsModalService.show.and.callThrough();
+            return mockBsModalService;
+          }
+        },
+        {
+          provide: SpaceService,
+          useFactory: (): jasmine.SpyObj<SpaceService> => {
+            const mockSpaceService: any = createMock(SpaceService);
+            mockSpaceService.getSpacesByUser.and.returnValue(of([spaceMock1, spaceMock2]));
+            mockSpaceService.deleteSpace = () => {};
+            return mockSpaceService;
+          }
+        },
+        {
+          provide: UserService,
+          useFactory: (): jasmine.SpyObj<UserService> => {
+            const mockUserService: jasmine.SpyObj<UserService> = createMock(UserService);
+            mockUserService.loggedInUser = of(mockUser) as ConnectableObservable<User> & jasmine.Spy;
```

Comment:
Does `of` really return a `ConnectableObservable`? Wouldn't you need to put a `.publish` on there or something?